hpx/runtime/threads/policies/lockfree_queue_backends.hpp

```diff
@@ -66,11 +85,17 @@ struct lockfree_fifo
     };
 };
 
+///////////////////////////////////////////////////////////////////////////////
 // LIFO
 template <typename T>
 struct lockfree_lifo_backend
 {
+#if defined(HPX_HAVE_CXX11_STD_ATOMIC_128BIT)
     typedef boost::lockfree::deque<T> container_type;
+#else
+    typedef boost::lockfree::queue<T> container_type;
```

Comment:
Shouldn't this be `boost::lockfree::stack`?

Reply:
Good point, will change.